fission/src/systems/simulation/synthesis_brain/SynthesisBrain.ts

```diff
@@ -1,39 +1,47 @@
-import Mechanism from "@/systems/physics/Mechanism";
-import Brain from "../Brain";
-import Behavior from "../behavior/Behavior";
-import World from "@/systems/World";
-import WheelDriver from "../driver/WheelDriver";
-import WheelRotationStimulus from "../stimulus/WheelStimulus";
-import ArcadeDriveBehavior from "../behavior/ArcadeDriveBehavior";
-import { SimulationLayer } from "../SimulationSystem";
-import Jolt from "@barclah/jolt-physics";
-import JOLT from "@/util/loading/JoltSyncLoader";
-import HingeDriver from "../driver/HingeDriver";
-import HingeStimulus from "../stimulus/HingeStimulus";
-import GenericArmBehavior from "../behavior/GenericArmBehavior";
-import SliderDriver from "../driver/SliderDriver";
-import SliderStimulus from "../stimulus/SliderStimulus";
-import GenericElevatorBehavior from "../behavior/GenericElevatorBehavior";
-import InputSystem, { AxisInput } from "@/systems/input/InputSystem";
-import DefaultInputs from "@/systems/input/DefaultInputs";
+import Mechanism from "@/systems/physics/Mechanism"
+import Brain from "../Brain"
+import Behavior from "../behavior/Behavior"
+import World from "@/systems/World"
+import WheelDriver from "../driver/WheelDriver"
+import WheelRotationStimulus from "../stimulus/WheelStimulus"
+import ArcadeDriveBehavior from "../behavior/ArcadeDriveBehavior"
+import { SimulationLayer } from "../SimulationSystem"
+import Jolt from "@barclah/jolt-physics"
+import JOLT from "@/util/loading/JoltSyncLoader"
+import HingeDriver from "../driver/HingeDriver"
+import HingeStimulus from "../stimulus/HingeStimulus"
+import GenericArmBehavior from "../behavior/GenericArmBehavior"
+import SliderDriver from "../driver/SliderDriver"
+import SliderStimulus from "../stimulus/SliderStimulus"
+import GenericElevatorBehavior from "../behavior/GenericElevatorBehavior"
+import { AxisInput, ButtonInput, Input } from "@/systems/input/InputSystem"
+import DefaultInputs, { InputScheme } from "@/systems/input/DefaultInputs"
+import PreferencesSystem from "@/systems/preferences/PreferencesSystem"
 
 class SynthesisBrain extends Brain {
     private _behaviors: Behavior[] = []
     private _simLayer: SimulationLayer
 
     // Tracks how many joins have been made for unique controls
-    private _currentJointIndex = 1;
+    private _currentJointIndex = 1
 
-    private static _currentRobotIndex = 0;
-
-    // Tracks how many robots are spawned for control identification
     private _assemblyName: string
+    private _assemblyIndex: number = 0
 
-    public constructor(mechanism: Mechanism, assemblyName: string) {
-        super(mechanism);
+    // Tracks the number of each specific mira file spawned
+    public static numberRobotsSpawned: { [key: string]: number } = {}
```

Comment:
Could this and `robotsSpawned` be combined? Maybe a map from key to list of spawned robots? That might be a bit slower if we're iterating over `robotsSpawned` a lot but it would also mean the information would only be stored in one place and we don't have to maintain two separate places.

Reply:
An issue I see with this is if you spawn two robots for example 'mean machine [0]' and 'mean machine [1]', then remove only 'mean machine [0]', it needs to know that the robot still around is index 1 not 0. I can't think of an easy way around this, but let me know if you do.